byterun/str.c

```diff
@@ -411,7 +411,7 @@ CAMLexport value caml_alloc_sprintf(const char * format, ...)
        Note that caml_alloc_string left room for a '\0' at position n,
        so the size passed to vsnprintf is n+1. */
     va_start(args, format);
-    vsnprintf(String_val(res), n + 1, saved_format, args);
+    vsnprintf((char *)String_val(res), n + 1, saved_format, args);
```

Comment:
The common pattern I'm looking for during review is that when you've got a `(char *)` cast, the thing being written into is a freshly-allocated string.  It looks like that is true---but maybe there should be a new `caml_alloc_writeable_string` function or similar which you could use instead, returning `(char*)`, to avoid these casts?